extruct/uniform.py

```diff
@@ -1,16 +1,28 @@
 from six.moves.urllib.parse import urlparse, urljoin
 
 
-def _uopengraph(extracted):
+def _uopengraph(extracted, with_og_arr=False):
     out = []
     for obj in extracted:
-        # In order of appearance in the page
         properties = list(reversed(obj['properties']))
-        # Ensuring that never empty value is returned if there is a duplicated
-        # property with non empty value
+        # Set of non empty properties
         non_empty_props = {k for k, v in properties if v and v.strip()}
+        # Set of repeated properties with at least 2 non empty values
+        repeated_props = {}
+        if with_og_arr:
+            repeated_props = {k for k in non_empty_props
+                              if len([i for i, v in properties if i == k and (v and v.strip())]) > 1}
```

Comment:
I think it's possible to be more efficient here, right now this is O(n**2) because we loop over all properties for all non-empty properties, while it should be possible to build a dict with repeated properties and their values in one go I think.

Reply:
Okay yeah you're right this is very inefficient will change